### PR TITLE
Loanable sequence template [10090]

### PR DESCRIPTION
**Pull request by MiguelCompany**
This PR adds basic support for loanable sequences, which are required for the implementation of DDS compliant read / take

Iterators support is still pending, but we can add them on a follow-up PR.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3238/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/449/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2895/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3270/)

**Review by IkerLuengo — APPROVED** [review]
LGTM.

Please, can you clarify if my understanding on the LoanableCollection::loan is correct?

File: include/fastdds/dds/core/LoanableCollection.hpp [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+     * @post has_ownership() == false
+     * @post maximum() == new_maximum
+     * @post length() == new_length
+     */
+    bool loan(
+            element_type* buffer,
+            size_type new_maximum,
+            size_type new_length)
+    {
+        if (has_ownership_ && maximum_ > 0)
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply include/fastdds/dds/core/LoanableCollection.hpp:143]: Correct 😉

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3252/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/463/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2905/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3284/)

**Review by IkerLuengo — CHANGES_REQUESTED** [review]

**Review by IkerLuengo — DISMISSED** [review]
LGTM

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3278/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/489/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2929/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3310/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3290/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/501/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2937/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3322/)

**Review by IkerLuengo — APPROVED** [review]
👍🏼

**Pull request merged**